openfl/federated/data/loader_keras.py

```diff
@@ -40,10 +40,33 @@ def __init__(self, batch_size, **kwargs):
     def get_feature_shape(self):
         """Returns the shape of an example feature array.
 
+        This method handles cases where no data is loaded (minimal initialization).
+        For MNIST-like datasets, it returns [28, 28, 1] (channels last format).
+
+        Returns:
+            list: The shape of an example feature array.
+        """
+        if self.X_train is not None and len(self.X_train) > 0:
+            # Return actual shape from loaded data if available
+            return list(self.X_train[0].shape)
+        # For minimal initialization, return a default MNIST-like shape
+        # Child classes should override this method if they work with different data shapes
+        return [28, 28, 1]  # Default Keras shape: [height, width, channels]
```

Comment:
All of these hardcoded values for feature shape / num classes will cause unexpected behavior for models where feature shape is intended to be set dynamically. It might be better to set this to `None` to force the user to overwrite this in their examples / existing OpenFL workspace. Otherwise this could lead to the model being initialized, then seeing a mismatch during the first round of an experiment that becomes more difficult to debug.

Reply:
+1
@rahulga1 
Related: https://github.com/securefederatedai/openfl/pull/1590#discussion_r2081435260